### PR TITLE
Refactor dropdown buttons

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
- Moved core button functionality to Button.js from ActionView.js
 - Refactored OverlayActionListView to use the new button tag
 - Added keyboard navigation to OverlayDropdown and OverlayActionListView to improve accessibility
 - LanguageChoiceView and UserNavigationInfoView now use OverlayActionListView in order to display menus

### CODE REVIEW

File: src/foam/u2/view/OverlayActionListView.js [hunk 1]
```diff
@@ -7,18 +7,21 @@
 foam.CLASS({
   package: 'foam.u2.view',
   name: 'OverlayActionListView',
-  extends: 'foam.u2.View',
+  extends: 'foam.u2.tag.Button',
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/OverlayActionListView.js:10]: Another approach could be to have a generic dropdownButtonView that this view extends as this one is restricted to adding only actions to the overlay but that could be done once we use the overlay/dropdown button for more things

File: src/foam/u2/view/OverlayActionListView.js [hunk 2]
```diff
@@ -7,18 +7,21 @@
...
     'foam.core.ConstantSlot',
     'foam.core.ExpressionSlot',
-    'foam.u2.md.OverlayDropdown'
+    'foam.u2.md.OverlayDropdown',
+    'foam.u2.HTMLView'
   ],
 
   imports: [
-    'ctrl'
+    'ctrl',
```
Comment (kgrgreer) [inline src/foam/u2/view/OverlayActionListView.js:22]: sort imports

File: src/foam/u2/view/OverlayActionListView.js [hunk 3]
```diff
@@ -216,12 +177,36 @@ foam.CLASS({
...
-      this.hovered_ = false;
+    function onKeyDown(e) {
+      var isTabPressed = (e.key === 'Tab' || e.keyCode === 9);
+
+      if ( ! isTabPressed ) {
+        return;
+      }
+
+      if ( e.shiftKey ) {
+        if ( this.document.activeElement === this.document.getElementById(this.firstEl_.id) ) {
```
Comment (kgrgreer) [inline src/foam/u2/view/OverlayActionListView.js:200]: Is this not the same as this.firstEl_.el()?

Comment (sarthak-marwaha) [reply src/foam/u2/view/OverlayActionListView.js]: It is. Didn't know we had that function. Fixed

**Comment by kgrgreer** [comment]
Has merge conflicts.

**Comment by sarthak-marwaha** [comment]
@kgrgreer fixed merge conflicts

**Pull request merged**